tensorflow/java/src/main/java/org/tensorflow/op/AbstractSingleOp.java

```diff
@@ -0,0 +1,60 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+package org.tensorflow.op;
+
+import org.tensorflow.Operation;
+
+/** A base class for an {@link Op} that wraps a single operation. */
+public abstract class AbstractSingleOp implements Op {
```

Comment:
Pretty much the same thing here, this class basically ensures a common equals/hashcode/toString behaviour among single operations, including custom ones.